src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/ProxyHelper.java

```diff
@@ -51,6 +51,18 @@ public ProxyHelper(Map<String, String> env) {
    */
   public Proxy createProxyIfNeeded(URL requestedUrl) throws IOException {
     String proxyAddress = null;
+    String noProxyUrl = env.get("no_proxy");
+    if (Strings.isNullOrEmpty(noProxyUrl)) {
+      noProxyUrl = env.get("NO_PROXY");
+    }
+    String[] noProxyUrlArray = noProxyUrl.split(",");
+    URL url = new URL(requestedUrl);
+    String requestedHost = url.getHost();
+    for(int i=0;i<noProxyUrlArray.length;i++){
+      if (requestedHost.contains(noProxyUrlArray[i])){
```

Comment:
I think you need a `!`.  Say I have `requestUrl = URL("http://mybiz.com")` and `export NO_PROXY=mybiz.com`.  Then then this _would_ create a proxy, right?

Also, I think you need to use `endsWith`, not `contains`.   According to https://www.gnu.org/software/wget/manual/html_node/Proxies.html:

> *no_proxy*
> This variable should contain a comma-separated list of domain extensions proxy should not be used for. For instance, if the value of no_proxy is ‘.mit.edu’, proxy will not be used to retrieve documents from MIT.

In the example above, suppose `NO_PROXY=biz`, then it wouldn't use a proxy for http://mybiz.com.

Finally, please indent the contents of the if by 2 spaces and add spaces around the for-loop's =, <, and after each ; (to match the coding style of the rest of the file).